pkg/tuned/dialog.js

```diff
@@ -122,46 +122,34 @@ define([
             var tuned;
             var dialog_selected;
 
-            function set_profile(set_progress_message) {
-                var dfd = $.Deferred();
-
+            function set_profile() {
                 // no need to check input here, all states are valid
-                set_progress_message(_("Switching performance profile"));
-
                 var profile = dialog_selected;
                 if (profile == "none") {
-                    tuned.call('/Tuned', 'com.redhat.tuned.control', 'stop', [])
-                        .done(function() {
+                    return tuned.call('/Tuned', 'com.redhat.tuned.control', 'stop', [])
+                        .then(function() {
                             update_button();
-                            dfd.resolve();
-                        })
-                        .fail(function(ex) {
-                            dfd.reject(ex);
+                            return null;
                         });
                 } else {
-                    tuned.call('/Tuned', 'com.redhat.tuned.control', 'switch_profile', [ profile ])
+                    return tuned.call('/Tuned', 'com.redhat.tuned.control', 'switch_profile', [ profile ])
                         .then(function(results) {
                             if (!results[0][0]) {
-                                dfd.reject(results[0][1] || _("Failed to switch profile"));
+                                return cockpit.reject(results[0][1] || _("Failed to switch profile"));
```

Comment:
Unfortunately cockpit.reject() doesn't exist yet. I'll make a pull request that adds it.